pkg/logger/logger.go

```diff
@@ -93,7 +93,7 @@ func NewNoopLogger() *ZapLogger {
 	}
 }
 
-func NewLogger(logFormat, logLevel string) (*ZapLogger, error) {
+func NewLogger(logFormat, logLevel, logTimestampFormat string) (*ZapLogger, error) {
```

Comment:
We should refactor this signature to use the options builder pattern and if an option is not provided we just default it.

For example,

```go
NewLogger(
  WithFormat("text"),
  WithLevel("info"),
  WithTimestampFormat("ISO8601"),
)
```

Reply:
Added issue #1355 to follow up with this suggestion.